modules/plugins/session/nvim-session-manager/config.nix

```diff
@@ -31,7 +32,10 @@ in {
       pluginRC.nvim-session-manager = entryAnywhere ''
         local Path = require('plenary.path')
         local sm = require('session_manager.config')
-        require('session_manager').setup(${toLuaObject cfg.setupOpts})
+        require('session_manager').setup(${toLuaObject (cfg.setupOpts
+          // {
+            autoload_mode = mkLuaInline "sm.AutoloadMode.${cfg.setupOpts.autoload_mode}";
+          })})
```

Comment:
Is there anything stopping us from setting this in `setupOpts` as a default?

Reply:
Would this not force users to also include `sm.AutoloadMode.` when setting `vim.session.nvim-session-manager.setupOpts.autoload_mode`?

Reply:
Current code setting this value, for reference:
https://github.com/NotAShelf/nvf/blob/6b6e7a0cad105f97553d86686c0859826630504e/modules/plugins/session/nvim-session-manager/nvim-session-manager.nix#L69-L73

Reply:
Oh, I just realised that `mkOption` has an `apply` argument that transforms the value before use: https://noogle.dev/f/lib/mkOption

Should have looked into that first.

Reply:
It's using apply now, though I'm a big fan of the "sm" magic string, as it unclear that it relies on the variable defined in a different file:
https://github.com/NotAShelf/nvf/blob/98ced04bc01e5638c3c26ddf3837a55e12419bce/modules/plugins/session/nvim-session-manager/nvim-session-manager.nix#L72
https://github.com/NotAShelf/nvf/blob/98ced04bc01e5638c3c26ddf3837a55e12419bce/modules/plugins/session/nvim-session-manager/config.nix#L31-L35

Reply:
What would `defaultText` do here? Documentation mentions that it should be used if `default` cannot be rendered, but default is still a string. Or should `defaultText` be `"sm.AutoloadMode.LastSession"` to indicate the actual final value?

Reply:
I think having `defaulText` be `"sm.AutoloadMode.LastSession"` is confusing to users reading through the options in the manual, so I'll only include the comment.